src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,49 +8,82 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "block, some street, unit number, postalcode";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final Postal postal;
     private boolean isPrivate;
 
     /**
      * Validates given address.
+     * @param postalName 
+     * @param unitName 
+     * @param streetName 
      *
      * @throws IllegalValueException if given address string is invalid.
      */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
+    public Address(String blockName, String streetName, String unitName, String postalName, boolean isPrivate) {
+ 
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
+        this.block = new Block(blockName);
+        this.street = new Street(streetName);
+        this.unit = new Unit(unitName);
+        this.postal = new Postal(postalName);
+        this.value = block.getName()+" ,"+street.getName()+" ,"+unit.getName()+" ,"+postal.getName();
+        
     }
 
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
+   /* public Address(String value, boolean isPrivate) {
+        String[] fullAddress = value.split("'");
+        this.block = new Block(fullAddress[0]);
+        this.street = new Street(fullAddress[1]);
+        this.unit = new Unit(fullAddress[2]);
+        this.postal = new Postal(fullAddress[3]);
+        
+    } */
+
 
     @Override
     public String toString() {
-        return value;
+        String fullAddress = block.getName() + ", "+ street.getName() +", " + unit.getName() + ", " + postal.getName();
+        return fullAddress;
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.block.equals(((Address) other).block) // state check
+                && this.street.equals(((Address) other).street)
+                && this.unit.equals(((Address) other).unit)
+                && this.postal.equals(((Address) other).postal));
     }
 
     @Override
     public int hashCode() {
         return value.hashCode();
     }
+    
+    public Block getBlock() {
```

Comment:
think why would you want to expose the block object outside the address class (at least in this context); in case you need to update any of the address fields, isn't it better to have methods to do that specifically rather than return the respective objects?